### PR TITLE
feat(xpath): move cypress-xpath to @cypress/xpath

### LINKED ISSUE #180: Moving to Core Cypress Repo
In an effort to better maintain and enhance this official plugin, we will be moving it into the [main Cypress repo](https://github.com/cypress-io/cypress) sometime in the next few weeks. During that time, we will be going through issues and PRs to make sure all relevant items are moved over and labeled appropriately.

Once it is moved, this repo will be archived and will become read-only.

### PR DESCRIPTION
**Pull request by jordanpowell88**
- Closes https://github.com/cypress-io/cypress-xpath/issues/180

### User facing changelog
Moves `cypress-xpath` plugin into `@cypress/xpath`

### Additional details
In an effort to better maintain and enhance this official plugin, we will be moving it into the main Cypress repo. 

### Steps to test
`yarn workspace @cypress/xpath test`

### PR Tasks

- [X] Have tests been added/updated?
- [X] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [X] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: cli/package.json [hunk 1]
```diff
@@ -161,6 +162,10 @@
       "import": "./svelte/dist/cypress-svelte.esm-bundler.js",
       "require": "./svelte/dist/cypress-svelte.cjs.js",
       "types": "./svelte/dist/index.d.ts"
+    },
+    "./xpath": {
+      "require": "./xpath/scripts",
+      "types": "./xpath/src"
```
Comment (emilyrohrbough) [inline cli/package.json]: This change indicated this will ship this first-class with each Cypress release as well.  I didn't realize as we are migrating to the mono-repo -- was this change intentional?

File: npm/xpath/package.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{
+  "name": "@cypress/xpath",
+  "version": "0.0.0-development",
+  "description": "Adds XPath command to Cypress.io test runner",
+  "main": "scripts",
+  "scripts": {
+    "test": "node ../../scripts/cypress.js run --e2e",
+    "cy:open": "node ../../scripts/cypress.js open --e2e --project ${PWD}",
```
Comment (emilyrohrbough) [inline npm/xpath/package.json]: nit: name these cypress:open and cypress:run to align with other cypress test scripts in the repo

File: npm/xpath/package.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    "src"
+  ],
+  "types": "src",
+  "license": "MIT",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/cypress-io/cypress.git"
+  },
+  "homepage": "https://github.com/cypress-io/cypress/tree/main/npm/xpath#readme",
+  "author": "Jordan <jordan@jpdesigning.com>",
```
Comment (emilyrohrbough) [inline npm/xpath/package.json]: Shouldn't this be "Cypress" or "Cypress Tools Team"?

File: npm/xpath/package.json [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  "files": [
+    "src"
+  ],
+  "types": "src",
+  "license": "MIT",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/cypress-io/cypress.git"
+  },
+  "homepage": "https://github.com/cypress-io/cypress/tree/main/npm/xpath#readme",
```
Comment (emilyrohrbough) [inline npm/xpath/package.json]:
```suggestion
  "homepage": "https://github.com/cypress-io/cypress/tree/develop/npm/xpath#readme",
```

File: npm/xpath/cypress/support/e2e.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+//
+// You can change the location of this file or turn off
+// automatically serving support files with the
+// 'supportFile' configuration option.
+//
+// You can read more here:
+// https://on.cypress.io/configuration
+// ***********************************************************
+
+import '../../src'
```
Comment (emilyrohrbough) [inline npm/xpath/cypress/support/e2e.js:16]: The video recording [`npm/xpath/cypress/videos/spec.cy.js.mp4`](https://github.com/cypress-io/cypress/pull/23878/files#diff-89a77b6c2f53a8fb8646c230085542440fcd82842c3627e7a14620a79cdafc65)should be deleted. & need to update the root .gitignore to ignore this so they aren't checked in to github on accident

File: npm/xpath/README.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+
+
+For more, see [Intelligent Code Completion](https://on.cypress.io/intellisense)
+
+## Roadmap
+
+- [x] wrap returned DOM nodes in jQuery [#2](https://github.com/cypress-io/cypress-xpath/issues/2)
+- [x] retry the assertion that follows [#3](https://github.com/cypress-io/cypress-xpath/issues/3)
+- [x] add TypeScript definitions [#4](https://github.com/cypress-io/cypress-xpath/issues/4)
+- [ ] search from the previous subject element [#5](https://github.com/cypress-io/cypress-xpath/issues/5)
```
Comment (emilyrohrbough) [inline npm/xpath/README.md]: can we transfer issue 5 to this repo and update this link?

File: cli/package.json (hunk 1 above)
Comment (admah) [reply cli/package.json]: @emilyrohrbough yes, part of my team's Q3 OKRs was to move all of our official plugins ([listed here](https://github.com/cypress-io/prod-eng-docs/blob/main/team-cypress-tools/charter.md#functional-responsibilities)) into the core mono-repo. As a result, we're also helping on Firewatch and adding appropriate labels (ie. state: routed to tools) for triage. 

For a little more clarity, the official plugins that we manage **are not** part of the actual cypress binary. They are releases as a separate package when needed like @cypress/schematic and others.

Comment (emilyrohrbough) [reply cli/package.json]: @admah This change also ships this plugin with the `cypress` CLI in addition to moving to the mono-repo. Just wanted to confirm we were going to support usage for both:
1. pulling from the `cypress` module
```
require('cypress/xpath')
```
2. pulling from standalone dependency
```
require('@cypress/xpath');
```

Comment (jordanpowell88) [reply cli/package.json]: Good point @emilyrohrbough . I will remove it from the CLI syncing

Comment (admah) [reply cli/package.json]: Yeah, good catch @emilyrohrbough. That is not what we want currently.

File: npm/xpath/README.md (hunk 6 above)
Comment (jordanpowell88) [reply npm/xpath/README.md]: Issue 5 is over 4 years old and is not something the tools team intends to do. I'm just going to remove this section since it's the only issue remaining

**Review by admah — APPROVED** [review]
lgtm 🚀

**Comment by emilyrohrbough** [comment]
@jordanpowell88 Did you need to configured the semantic release rules?  https://github.com/cypress-io/cypress/blob/develop/npm/angular/.releaserc.js

**Comment by admah** [comment]
Ah....I forgot about that step @emilyrohrbough. @jordanpowell88 there's no diff for that file between internal and external packages. You should just recreate it in the package folder and be good to go.

**Comment by emilyrohrbough** [comment]
@admah I had too 😅

**Review by emilyrohrbough — APPROVED** [review]

File: npm/xpath/.releaserc.js [hunk 7]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+  ...require('../../.releaserc.base'),
+  branches: [
+    { name: 'master', channel: 'latest' },
```
Comment (emilyrohrbough) [inline npm/xpath/.releaserc.js]:
```suggestion
    { name: 'develop', channel: 'latest' },
```
Since https://github.com/cypress-io/cypress/pull/23899 merged, this needs updated
